Content.Client/Examine/ExamineSystem.cs

```diff
@@ -90,40 +89,97 @@ private bool HandleExamine(in PointerInputCmdHandler.PointerInputCmdArgs args)
         }
 
-        public async void DoExamine(EntityUid entity)
+        private void OnExamineInfoResponse(ExamineSystemMessages.ExamineInfoResponseMessage ev)
+        {
+            var player = _playerManager.LocalPlayer?.ControlledEntity;
+            if (player == null)
+                return;
+
+            // Tooltips coming in from the server generally prioritize
+            // opening at the old tooltip rather than the cursor/another entity,
+            // since there's probably one open already if it's coming in from the server.
+            OpenTooltip(player.Value, ev.EntityUid, ev.CenterAtCursor, ev.OpenAtOldTooltip);
+            UpdateTooltipInfo(player.Value, ev.EntityUid, ev.Message, ev.GetVerbs);
+        }
+
+        private void OnVerbsResponse(VerbsResponseEvent ev)
+        {
+            if (ev.Verbs == null || _examineTooltipOpen == null)
+                return;
+
+            // Get examine verbs then add them all to the tooltip
+            if (ev.Verbs.TryGetValue(VerbType.Examine, out var set))
+            {
+                AddVerbsToTooltip(set);
+            }
+        }
+
+        public override void SendExamineTooltip(EntityUid player, EntityUid target, FormattedMessage message, bool getVerbs, bool centerAtCursor)
+        {
+            OpenTooltip(player, target, centerAtCursor, false);
+            UpdateTooltipInfo(player, target, message, getVerbs);
+        }
+
+        /// <summary>
+        ///     Opens the tooltip window and sets spriteview/name/etc, but does
+        ///     not fill it with information. This is done when the server sends examine info/verbs,
+        ///     or immediately if it's entirely clientside.
+        /// </summary>
+        public void OpenTooltip(EntityUid player, EntityUid target, bool centeredOnCursor=true, bool openAtOldTooltip=true)
         {
             // Close any examine tooltip that might already be opened
+            // Before we do that, save its position. We'll prioritize opening any new popups there if
+            // openAtOldTooltip is true.
+            var oldTooltipPos = _examineTooltipOpen?.ScreenCoordinates;
             CloseTooltip();
 
             // cache entity for Update function
-            _examinedEntity = entity;
+            _examinedEntity = target;
 
             const float minWidth = 300;
-            var popupPos = _userInterfaceManager.MousePositionScaled;
+            ScreenCoordinates popupPos;
+
+            if (openAtOldTooltip && oldTooltipPos != null)
+            {
+                popupPos = oldTooltipPos.Value;
+            }
+            else if (centeredOnCursor)
+            {
+                popupPos = _userInterfaceManager.MousePositionScaled;
+            }
+            else
+            {
+                popupPos = _eyeManager.CoordinatesToScreen(Transform(target).Coordinates);
+            }
 
             // Actually open the tooltip.
             _examineTooltipOpen = new Popup { MaxWidth = 400 };
```

Comment:
maybe a bit out of scope, but all the _examineTooltip layout stuff should probably just be in xaml somewhere. this should just be opening & setting the label text & sprite-view's sprite.